cdap-ui/app/cdap/services/datasource/index.js

```diff
@@ -115,9 +123,20 @@ export default class Datasource {
     });
     this.eventEmitter.on(WINDOW_ON_FOCUS, this.resumePoll.bind(this));
     this.eventEmitter.on(WINDOW_ON_BLUR, this.pausePoll.bind(this));
+    if (!this.excludeFromHealthCheck) {
+      SystemDelayStore.dispatch({
+        type: SystemDelayActions.registerDataSource,
+        payload: this,
+      });
+    }
+  }
+
+  getBindings(){
+    return cloneDeep(Object.values(this.bindings));
```

Comment:
is cloneDeep required here? should avoid using cloneDeep

Reply:
Using this so we don't accidentally update anything in the original bindings as we only need a snapshot of bindings when the health check happens